ugs-platform/DowelModule/src/main/java/com/willwinder/ugs/platform/dowel/DowelPreview.kt

```diff
@@ -58,7 +57,7 @@ class DowelPreview(val description: String, val generator: DowelGenerator) : Ren
     override fun reloadPreferences(vo: VisualizerOptions?) {
     }
 
-    override fun draw(drawable: GLAutoDrawable?, idle: Boolean, workCoord: Point3d?, objectMin: Point3d?, objectMax: Point3d?, scaleFactor: Double, mouseWorldCoordinates: Point3d?, rotation: Point3d?) {
+    override fun draw(drawable: GLAutoDrawable, idle: Boolean, machineCoord: Point3d, workCoord: Point3d, objectMin: Point3d, objectMax: Point3d, scaleFactor: Double, mouseWorldCoordinates: Point3d, rotation: Point3d) {
```

Comment:
The question marks here are still valid, did this cause a build error?

Reply:
It means the value might be null, since the call comes from Java where the null safety features don't exist.

Reply:
This was unintentional, I used the refactoring functions in IntelliJ. It wasn't smart enough to handle Kotlin, and I was too distracted to go through these changes (it's usually ok). Now I know better.